File: app-mvp/src/main/java/com/test/app_mvp/contract/RepositoryListContract.java
package com.test.app_mvp.contract;

import com.test.app_mvp.model.GitHubService;

/**
 * Created by justluvher on 2018. 5. 30..
 */

public interface RepositoryListContract {

    //mvp에서 activity가 구현할 view 인터페이스
    //뷰을 업데이트 한다.
    interface View {
        String getSelectedLanguage();
        void showProgress();
        void hideProgress();
        void showRepositories(GitHubService.Repositories repositories);
        void showError();
        void startDetailActivity(String fullRepositoryName);
    }

    //activity에서 user 이벤트 인터페이스
    interface UserActions {
        void selectLanguage(String language);
        void selectRepositoryItem(GitHubService.RepositoryItem item);
    }
}

File: app-mvp-kotlin/src/main/java/com/test/app_mvp_kotlin/contract/RepositoryListContract.kt
package com.test.app_mvp_kotlin.contract

/**
 * Created by justluvher on 2018. 6. 6..
 */

interface RepositoryListcontract {

    //Repository activty가 구현할 view 인터페이스를 정의한다.
    //뷰를 업데이트 한다.
    interface View {
        fun getSelectedLanguage(): String;
        fun showProgress();
        fun hideProgress();
        fun showRepositories();
        fun startDetailActivity(fullRepositoryName: String);
    }

    //Repository activity가 구현할 user 이벤트 인터페이스를 정의한다.
    interface UserActions {
        fun selectLanguage(language: String);
        fun selectRepositoryItem();
    }
}